test/integration/minishift.go

```diff
@@ -190,6 +190,47 @@ func (m *Minishift) executingMinishiftCommand(command string) error {
 	return nil
 }
 
+func (m *Minishift) setImageCaching(operation string) error {
+	enabled := "true"
+	if operation == "disabled" {
+		enabled = "false"
+	}
+
+	return m.executingMinishiftCommand(fmt.Sprintf("config set image-caching %s", enabled))
+}
+
+func (m *Minishift) imageExportShouldComplete(noOfImages int) error {
+	// poll till the output of the `minishift image list` shows number of cached images
+	timeout := time.NewTimer(20 * time.Minute)
+
+outerPollActive:
+	for {
+		select {
+		case <-timeout.C:
+			return errors.New("Timed out in getting the number of default cached images")
+		default:
+			cmdOut, _, _ := m.runner.RunCommand("image list")
+			cmdOut = strings.TrimRight(cmdOut, "\n")
+			numOfLines := len(strings.Split(cmdOut, "\n"))
+			if numOfLines == noOfImages {
+				break outerPollActive
+			}
+			if numOfLines > noOfImages {
```

Comment:
This is added in case, `minishift image list` shows 5 images upfront and we are expecting 3.

Reply:
I do not think this will be help full by counting the caching image number IMO. 

Consider this Usecase:
1. minishift start with default configuration (i mean image caching enabled by default) 
2. Export one image (minishift image export ...)
3. minishift delete
4. minishift start
What i am expecting here is that all four images will be cached, whereas the validation is done on number of lines i.e three.

Reply:
@amitkrout , 

> What i am expecting here is that all four images will be cached

This assumption is wrong here. Only openshift images will get automatically imported for next start. You need to import other images explicitely.

See the scenario https://github.com/budhrg/minishift/blob/issue-1799/test/integration/features/cmd-image.feature#L28-L51

Reply:
@amitkrout `noOfImages` is an argument to the function and the value comes from `s.Step(`^image export completes with (\d+) images$`, minishift.imageExportShouldComplete)` e.g. `And image export completes with 3 images`.So we are good here.

Reply:
@LalatenduMohanty @budhrg thanks for the explanation. Actually was little confused with import and export.